gocourse03/main.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"fmt"
+	"time"
+)
+
+type rodentType string
+
+const (
+	Beaver   rodentType = "beaver"
+	Hamster  rodentType = "hamster"
+	Squirrel rodentType = "squirrel"
+)
+
+type (
+	FromTo [2]*Sector
+
+	Movement struct {
+		Time time.Time
+		FromTo
+	}
+
+	Rodent struct {
+		ID        int
+		Type      rodentType
+		Movements []*Movement
+	}
+
+	Sector struct {
+		Name    string
+		Rodents []*Rodent
+	}
+
+	Maze struct {
+		Sectors []*Sector
+	}
+)
+
+func NewRodent(id int, t rodentType) *Rodent {
+	return &Rodent{id, t, nil}
```

Comment:
Так бажано не робити, краще вказувати явно назви полів:
```suggestion
	return &Rodent{ID: id, Type: t, Movements: nil}
```

Уяви, якщо ми додамо ще якесь поле типу поінтер, у нас буде менш читабельно:
```go
	return &Rodent{id, t, nil, nil}
```

А якщо додамо поле типу `int`, то взагалі можемо сплутати порядок полей і через це будуть баги.